tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh

```diff
@@ -94,8 +120,8 @@ fi
 
 if [ "${force_build_from_source}" == "true" ]; then
 	info "Build cloud-hypervisor from source as it's been request via the force_build_from_source flag"
-	build_clh_from_source
+	check_cached_cloud_hypervisor
 else
 	pull_clh_released_binary ||
-	(info "Failed to pull cloud-hypervisor released binary, trying to build from source" && build_clh_from_source)
+	(info "Failed to pull cloud-hypervisor released binary, trying to build from source" && check_cached_cloud_hypervisor)
```

Comment:
Let's double-check this is what we want:

1. try to pull the released binary (`pull_clh_released_binary`)
2. If the release binary doesn't exist because `$cloud_hypervisor_version` points to a commit (or tag) (or failed to download) then:
  2.1 check  `$cloud_hypervisor_version` exist in the cache. If not then:
  2.2. build from source `build_clh_from_source`

@fidencio @GabyCT ^